### DISCUSSION TITLE
Advantage of using podman

### DISCUSSION THREAD

**Discussion by deajan** [jdoss/ppngx#16] [General]
Hello,

Just tried your ppngx script which is great, works good on RHEL 9 ;)

Since `podman-compose exists, one can adapt the paperless-ngx original install script to run without too much hassle.
I am using podman whenever I can because of the EL ecosystem.

What's the advantage of your script over the original `docker-compose` setup ?
I do like systemd unit files too for containers, since it makes things "manageable" and easily monitorable.

Are there any other nice advantages I didn't realise ?
Best regards.

**Comment by jdoss** [comment]
Great glad to hear it works well on RHEL 9! 

As for your question about why use this script, here is my very subjective answer as to why I created this project.

I think podman-compose and docker-compose are great for development setups and let any type of end user stand up containerized software quickly, but I would never run something in production with either one of them. When I created this script podman-compose was still very buggy and IIRC it still does not support Podman Pods. I also do not use anything related to Docker because I have major issues with how Docker Inc. has run their company and treated Open Source. 

I also really like to use systemd to manage my containerized workloads as you mentioned it makes things more manageable. For example you can set complex dependency requirements between each service and I also like running all of my services inside Podman Pods to isolate everything easily. I run Paperless on my Fedora workstation and I use the ppngx script to make changes to my setup and then I convert everything over to systemd units and copy them into `${HOME}/.config/systemd/user` to run them with systemd. You could just edit the systemd units manually as well and not use the ppngx script too. 

I think it really comes down to personal preference to be honest. I find it easier to mess around with containerized workloads that have many services using a simple bash script and Podman vs using Docker and Docker Compose. It just gives me more flexibility to make changes that I want to have when running Paperless NGX. 

I hope my response helps answer your question. :)

**Reply by deajan** [reply]
Makes sense, and perfectly replies to my question ^^
Thanks for your time, and thanks for the script ;)
